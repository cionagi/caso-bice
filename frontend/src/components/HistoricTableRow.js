import React from 'react';
import PropTypes from 'prop-types'; // ES6

const HistoricTableRow = ({ historicRow }) => {
  const { date, valueFormat } = historicRow;
  return (
    <tr>
      <td>{date}</td>
      <td>{valueFormat}</td>
    </tr>
  );
};

HistoricTableRow.propTypes = {
  historicRow: PropTypes.object.isRequired,
};

export default HistoricTableRow;
