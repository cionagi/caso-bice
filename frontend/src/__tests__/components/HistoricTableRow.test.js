import React from 'react';
import { render } from '@testing-library/react';

import HistoricTableRow from '../../components/HistoricTableRow';

const DATA_FAKE = {
  date: '23-04-2020',
  valueFormat: '1.00 USD',
};

test('Link renders correctly', () => {
  const { asFragment } = render(<HistoricTableRow historicRow={DATA_FAKE} />);
  expect(asFragment()).toMatchSnapshot();
});
