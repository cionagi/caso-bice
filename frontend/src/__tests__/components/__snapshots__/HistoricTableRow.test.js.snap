// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<DocumentFragment>
  <tr>
    <td>
      23-04-2020
    </td>
    <td>
      1.00 USD
    </td>
  </tr>
</DocumentFragment>
`;
